reconcilers/resource.go

```diff
@@ -381,7 +384,11 @@ func (r *ResourceReconciler[T]) reconcileOuter(ctx context.Context, req Request)
 		}
 
 		// Suppress result. Let the informer discover the resource mutation and requeue. Requeueing
-		// now may result in re-processing a stale cache.
+		// now may result in re-processing a stale cache. Honor the requeue after if it was set.
```

Comment:
Re-reading the comment, this probably doesn't make sense to honor. I should rather update our tests to match. The reasoning here is that because the status was updated, a requeue will happen soon anyways, right?